lib/image-editor/class-image-editor-rotate.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Start: Include for phase 2
+ * Image Editor: Image_Editor_Rotate class
+ *
+ * @package gutenberg
+ * @since 7.x ?
+ */
+
+/**
+ * Image editor modifier abstract class.
+ */
+require_once __DIR__ . '/class-image-editor-modifier.php';
+
+/**
+ * Rotate image modifier.
+ */
+class Image_Editor_Rotate extends Image_Editor_Modifier {
+
+	/**
+	 * Angle to rotate by.
+	 *
+	 * @var integer
+	 */
+	private $angle = 0;
+
+	/**
+	 * Constructor.
+	 *
+	 * Will populate object properties from the provided arguments.
+	 *
+	 * @param float $angle Angle to rotate by.
+	 */
+	public function __construct( $angle ) {
+		$this->angle = $this->restrict_angle( intval( $angle, 10 ) );
+	}
+
+	/**
+	 * Update the image metadata with the modifier.
+	 *
+	 * @access public
+	 *
+	 * @param array $meta Metadata to update.
+	 * @return array Updated metadata.
+	 */
+	public function apply_to_meta( $meta ) {
+		$meta['rotate'] += $this->angle;
+		$meta['rotate']  = $this->restrict_angle( $meta['rotate'] );
+
+		return $meta;
+	}
+
+	/**
+	 * Apply the rotate modifier to the image
+	 *
+	 * @access public
+	 *
+	 * @param WP_Image_Editor $image Image editor.
+	 * @return bool|WP_Error True on success, WP_Error object or false on failure.
+	 */
+	public function apply_to_image( $image ) {
+		return $image->rotate( 0 - $this->angle );
+	}
+
+	/**
+	 * Puts the angle in the range [ 0, 360 ).
+	 *
+	 * @access private
+	 *
+	 * @param integer $angle Angle to restrict.
+	 * @return integer Restricted angle.
+	 */
+	private function restrict_angle( $angle ) {
+		if ( $angle >= 360 ) {
+			$angle = $angle % 360;
+		} elseif ( $angle < 0 ) {
+			$angle = 360 - ( abs( $angle ) % 360 );
+		}
+
+		return $angle;
```

Comment:
```suggestion
		return min ( $angle, 360 );
```
Make sure it less than 360?